charts/redpanda/templates/console/configmap-and-deployment.yaml

```diff
@@ -15,6 +15,28 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */}}
 {{ $values := .Values }}
+
+{{/* Secret */}}
+{{ $secretConfig := dict }}
+{{ if and .Values.console.enabled (not .Values.console.secret.create) }}
+{{ $licenseKey := ( include "enterprise-license" .  ) }}
+{{ $secretConfig = ( dict
+  "create" true
+  "enterprise" ( dict "license" $licenseKey)
+  )
+}}
+
+{{ $config := dict
+  "Values" (dict
+  "secret" $secretConfig
+  )}}
+
+{{ $console := deepCopy .Subcharts.console }}
+{{ $console = merge $config $console }}
+---
+{{ include (print .Subcharts.console.Template.BasePath "/secret.yaml") $console }}
```

Comment:
NIT: The file name `charts/redpanda/templates/console/configmap-and-deployment.yaml` does not reflect what this file will create. Now it is more of `configmap-secret-and-deployment.yaml`, right? Maybe to clear the confusion just rename the file to `console.yaml`. What do you think?

Reply:
That would be too much imho. Why not just break this up into different files? Is there any reason why not?

Reply:
The reason I merged those into one file is how helm templating engine works. When you define one variable it can not be referenced in another file. I might be wrong, but I could make it to work.

Reply:
You're right.

Reply:
We could break it up, but that would mean either moving the logic to a helper to avoid repetition, or copy/pasting everything.